docs/source/peer_ledger_snapshot.md

```diff
@@ -0,0 +1,153 @@
+
+**Note: snapshots can also be used to verify that the public state between peers is identical, even if no new peer will use the snapshot to join the channel. This can be useful in identifying if a ledger fork has occurred.**
+
+## Taking a snapshot
+
+For the full list of snapshot-related commands, check out [`peer snapshot` commands](./commands/peersnapshot.html).
+
+Before taking a snapshot, it is a best practice to confirm the current ledger height by issuing a command similar to:
+
+```
+peer channel getinfo -c <name of channel>
+```
+
+You will see a response similar to:
+
+```
+2018-02-25 15:15:44.135 UTC [channelCmd] InitCmdFactory -> INFO 003 Endorser and orderer connections initialized
+Blockchain info: {"height":970,"currentBlockHash":"JgK9lcaPUNmFb5Mp1qe1SVMsx3o/22Ct4+n5tejcXCw=","previousBlockHash":"f8lZXoAn3gF86zrFq7L1DzW2aKuabH9Ow6SIE5Y04a4="}
+2018-02-25 15:15:44.139 UTC [main] main -> INFO 006 Exiting.....
+```
+
+In this example, the ledger height is `970`.
+
+A snapshot request can be submitted by issuing a command similar to:
+
+```
+peer snapshot submitrequest -C <name of channel> -b <ledger height where snapshot will be taken> --peerAddress <address of peer> --tlsRootCertFile <path to root certificate of the TLS CA>
+```
+
+For example:
+
+```
+peer snapshot submitrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+**If you give a ledger height of `0`, the snapshot is taken immediately. This is useful for cases when an organization is interested in backing up its own ledger data and does not intend to share the data with another organization. Do not take these "immediate" snapshots in cases when snapshots will be evaluated from multiple peers, as it increases the likelihood that the snapshots will be taken at different ledger heights.**
+
+If the request is successful, you will see a `Snapshot request submitted successfully` message.
+
+You can list the pending snapshots by issuing a command similar to:
+
+```
+peer snapshot listpending -C testchannel --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+You will see a response similar to:
+
+```
+Successfully got pending snapshot requests [970]
+```
+
+When a snapshot has been generated for a particular block height, the pending request for that block height will no longer appear in the list. You can also verify that a snapshot has been create successfully by looking at the peer logs. You will see a message similar to `Generated snapshot block-<block number>`.
```

Comment:
I don't think that the actual log messages should go here as it's hard to maintain synergy between code and docs at this level of granularity. Instead, we can advise to look for the existence of the snapshot folder for the new snapshot.